### PR TITLE
Refine for min and max

### PR DESCRIPTION
**Pull request by rikardn**
Use some basic assumptions of positivity and negativity to refine `min` and `max`

### CODE REVIEW

File: symengine/refine.cpp [hunk 1]
```diff
@@ -70,6 +70,78 @@ void RefineVisitor::bvisit(const Conjugate &x)
...
+    vec_basic positive;
+    vec_basic keep;
+    bool have_negative = false;
+    bool have_nonpositive = false;
+    for (auto arg : x.get_args()) {
+        auto newarg = apply(arg);
+        if (is_true(is_negative(*newarg, assumptions_))) {
+            keep.push_back(newarg);
+            have_negative = true;
+        } else if (is_true(is_nonpositive(*newarg, assumptions_))) {
```
Comment (isuruf) [inline symengine/refine.cpp:123]: Can this be written as `is_false(is_negative())` and reuse `is_negative()`?

Comment (rikardn) [reply symengine/refine.cpp:123]: Thanks @isuruf for reviewing.   Yes, it should be possible.

Comment (rikardn) [reply symengine/refine.cpp:123]: This turns out to be problematic. Both `is_false(is_negative(expr))` and `is_false(is_nonnegative(expr))` can be true in the case where `expr` is non-real. I can go ahead with this change, but then I would need to check `is_real` first.

The option to this is to let `is_negative` etc throw for non-real numbers. My current idea for all `is_` has been to allow all numeric input and throw for `Sets`, `Logic` etc.

Comment (isuruf) [reply symengine/refine.cpp:123]: What does sympy do?

Comment (rikardn) [reply symengine/refine.cpp:123]: Sympy does not throw for non-reals:

```python
>>> import sympy
>>> sympy.I.is_positive
False
```

So either what we have in the PR now or first check `is_real` then `is_negative` results can be inverted to get `is_nonnegative` for free. What do you think @isuruf?

**Review by isuruf — APPROVED** [review]

**Comment by rikardn** [comment]
Thanks for reviewing

**Pull request merged**